public/scripts/controllers/booksShowController.js

```diff
@@ -0,0 +1,44 @@
+angular
+    .module('booksApp')
+    .controller('BooksShowController', BooksShowController);
+
+BooksShowController.$inject = ['$http', '$routeParams', '$location'];
+
+function BooksShowController($http, $routeParams, $location) {
+    var vm = this;
+    $http({
+        method: 'GET',
+        url: 'https://super-crud.herokuapp.com/books/' + $routeParams.id
+    }).then(function onSuccess(res) {
+        vm.book = res.data;
+    }, function onError(res) {
+        console.log('error', res);
+    });
+
+    vm.deleteBook = function(book) {
+        $http({
+            method: 'DELETE',
+            url: 'https://super-crud.herokuapp.com/books/' + book._id
+        }).then(function onSuccess(res) {
+            $location.path('/');
+        }, function onError(res) {
+            console.log('error', res);
+        });
+    }
+
+    vm.editBook = function(book) {
+        $http({
+            method: 'PUT',
+            url: 'https://super-crud.herokuapp.com/books/' + book._id,
+            data: book
+        }).then(function onSuccess(res) {
+            $location.path('/');
+        }, function onError(res) {
+            console.log('error', res);
+        });
+    }
+
+    vm.cancelEdit = function(book) {
```

Comment:
I don't think it is entirely necessary to have a cancel edit function since the user will stay on that page by default.  However, this would probably be a good place to include a feature that would repopulate the the title, author, release date, and img fields with the original text.